src/System.Private.CoreLib/src/System/Threading/Lock.cs

```diff
@@ -9,13 +9,17 @@ namespace System.Threading
 {
     public sealed class Lock
     {
+        // The following constants define characteristics of spinning logic in the Lock class
+        const int SpinningNotInitialized = 0;
+        const int SpinningDisabled = -1;
+        const int MaxSpinningValue = 10000;
```

Comment:
> trying to decipher what the intention of ”-1”

That's what the comments are for. The named constant for -1 is not making this better. The problem is that the code as written is making implicit assumptions about -1 to mean no spinning a few lines after `s_maxSpinCount` is initialized. If no spinning was defined say as MaxInt, things would break. To crack the code, you have to look at two unrelated places in the file. vs. if you see -1 right next to the place where it is taken advantage of, you can see all context locally without jumping around.

Similar implicit assumption is made about SpinningNotInitialized.

Defining 10000 as a named constant would be reasonable.

(Not asking you to fix this unless you really want to ... just explaining my POW.)